### PR TITLE
balabolka: Add version 2.15.0.751

### LINKED ISSUE #4460: [Request] Add Balabolka Text-To-Speech
Balabolka is an extensive freeware Text-To-Speech GUI program for Windows.

It interfaces with Microsoft Speech Platform and SAPI Text-To-Speech engines.
Such engines can be installed separately, but a few are included with Windows by default.

There does not seem to be any other Text-To-Speech software available via scoop.
I have been using Balabolka for years. It has active and committed developers.
It is the most popular alternative here: https://alternativeto.net/software/balabolka

website: http://balabolka.site/balabolka.htm
changelog: http://balabolka.site/changelog.txt
portable version: http://balabolka.site/balabolka_portable.zip

Shortcuts are needed for:
`balabolka.exe`

Settings and sessions of the portable version can be persisted via two plain text files.
These are not included in the .zip as they are generated and loaded at runtime, _they can be initially blank_.
`balabolka.pcfg`
`balabolka.dat`

### PR DESCRIPTION
**Pull request by erogy**
- Closes #4460

### CODE REVIEW

File: bucket/balabolka.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+    "version": "2.15.0.749",
+    "description": "Text-To-Speech software based on SAPI",
+    "homepage": "http://balabolka.site/balabolka.htm",
+    "license": "Freeware",
+    "url": "http://balabolka.site/balabolka_portable.zip",
+    "hash": "438b63a2deaa65b86a3f9d23a26b0dcae6500113c2ddd1058786cc7eed6f6b1e",
+    "extract_dir": "Balabolka",
+    "pre_install": "Get-ChildItem \"$persist_dir\\*\" -Include 'balabolka.dat', 'balabolka.pcfg' | Copy-Item -Destination \"$dir\" -ErrorAction SilentlyContinue }",
```
Comment (Ash258) [inline bucket/balabolka.json:9]: Why do you not test clean installation/update/uninstallation before pushing??

File: bucket/balabolka.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "uninstaller": {
+        "script": "Get-ChildItem \"$dir\\*\" -Include 'balabolka.dat', 'balabolka.pcfg' | Copy-Item -Destination \"$persist_dir\" -ErrorAction SilentlyContinue -Force }"
```
Comment (Ash258) [inline bucket/balabolka.json:11]: Above

File: bucket/balabolka.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{
+    "version": "2.15.0.749",
+    "description": "Text-To-Speech software based on SAPI",
+    "homepage": "http://balabolka.site/balabolka.htm",
+    "license": "Freeware",
+    "url": "http://balabolka.site/balabolka_portable.zip",
+    "hash": "438b63a2deaa65b86a3f9d23a26b0dcae6500113c2ddd1058786cc7eed6f6b1e",
+    "extract_dir": "Balabolka",
+    "pre_install": "if (Test-Path \"$persist_dir\") { Get-ChildItem \"$persist_dir\\*\" -Include 'balabolka.dat', 'balabolka.pcfg' | Copy-Item -Destination \"$dir\" -ErrorAction SilentlyContinue }",
```
Comment (Ash258) [inline bucket/balabolka.json:9]:
```suggestion
    "post_install": "Get-ChildItem \"$persist_dir\\*\" -Include 'balabolka.dat', 'balabolka.pcfg' | Copy-Item -Destination \"$dir\" -ErrorAction SilentlyContinue",
```

**Comment by erogy** [comment]
> What is balabolka.echo file??

It stores strings that are part of the settings and should also be persisted, I overlooked it. Also files `balabolka.pcfg` and `balabolka.dat` are hidden so `Get-ChildItem` needs the `-Force` switch to see them.

**Pull request merged**